WS2012R2/lisa/remote-scripts/ica/perf_ntttcp_client.sh

```diff
@@ -510,28 +510,135 @@ if [ -d  $HOME/$log_folder ]; then
+        sts=$?
+
+        current_tx_bytes=$(get_tx_bytes "$ETH_NAME")
+        current_tx_pkts=$(get_tx_pkts "$ETH_NAME")
+        bytes_new=$((current_tx_bytes - previous_tx_bytes))
+        pkts_new=$((current_tx_pkts - previous_tx_pkts))
+        avg_pkt_size=$(echo "scale=2;$bytes_new/$pkts_new/1024" | bc)
+        Throughput=$(echo "scale=2;$bytes_new/$TEST_DURATION*8/1024/1024/1024" | bc)
+        previous_tx_bytes=$current_tx_bytes
+        previous_tx_pkts=$current_tx_pkts
+
+        echo "Throughput (gbps): $Throughput"
+        echo "average packet size: $avg_pkt_size"
+        printf "%4s  %8.2f  %8.2f\\n" "${current_test_threads}" "${Throughput}" "${avg_pkt_size}" >> "${eth_log}"
+
+        echo "current test finished. wait 10 seconds for next one... "
+        sleep 10
+        pkill -x ntttcp
+        pkill -x lagscope
+        pkill -f ESTA
+        pkill -x sar
+        pkill -x dstat
+        pkill -x mpstat
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "pkill -x ntttcp"
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "pkill -x lagscope"
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "pkill -x sar"
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "pkill -x dstat"
+        ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+         "pkill -x mpstat"
+    fi
+    ethtool -S eth1 > "$HOME/$log_folder/sender-ethtool-eth1.log"
+    ethtool -S enP2p0s2 > "$HOME/$log_folder/sender-ethtool-enP2p0s2.log"
+    ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+     "ethtool -S eth1 > $HOME/$log_folder/receiver-ethtool-eth1.log"
+    ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2} \
+     "ethtool -S enP2p0s2 > $HOME/$log_folder/receiver-ethtool-enP2p0s2.log"
+    cat /proc/interrupts > "$HOME/$log_folder/sender-interrupts-end.log"
+    ssh -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -o StrictHostKeyChecking=no ${SERVER_OS_USERNAME}@${STATIC_IP2}  \
+    "cat /proc/interrupts > $HOME/$log_folder/receiver-interrupts-end.log"
+    LogMsg "Ntttcp UDP succeeded with all connections."
+    echo "Ntttcp  UDP succeeded with all connections." >> ~/summary.log
+    cd $HOME || exit
+   if ! scp -i "$HOME/.ssh/${SSH_PRIVATE_KEY}" -o StrictHostKeyChecking=no \
+    ${SERVER_OS_USERNAME}@[${STATIC_IP2}]:/root/"$log_folder"/* /root/"$log_folder"; then
+        echo "ERROR: Unable to transfer server side logs."
+        UpdateTestState $ICA_TESTFAILED
+    fi
+if [[ "${TEST_TYPE:-UNDEFINED}" = "UNDEFINED" ]] || [[ "${TEST_TYPE}" = "ntttcp" ]]; then
+    eth_log="$HOME/$log_folder/eth_report.log"
```

Comment:
as you can see, this is duplicated from above, and that's why it's better to be put at the top